packages/themed-components-stories/stories/utils/useData.ts

```diff
@@ -0,0 +1,174 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { useState, useEffect, useMemo, useCallback } from 'react'
+import {
+	CommunityData,
+	EdgeData,
+	JoinData,
+	LocalEntity,
+	NeighborLocalEntity,
+	NodeData,
+} from './types'
+import { loadRemoteData, search } from './utils'
+import { ICommunityDetail } from '@essex-js-toolkit/hierarchy-browser'
+
+export function useData(
+	selectedOption: string,
+): [
+	ICommunityDetail[],
+	LocalEntity[],
+	NeighborLocalEntity[],
+	(selection: string) => JoinData[][] | undefined,
+] {
+	const [nodes, setNodes] = useState<NodeData[] | undefined>()
+	const [edges, setEdges] = useState<EdgeData[] | undefined>()
+	const [join, setJoin] = useState<JoinData[] | undefined>()
+
+	const clusterIdMap = useMemo((): { [x: string]: JoinData[] } | undefined => {
+		if (join) {
+			return join.reduce((acc: { [x: string]: JoinData[] }, d: JoinData) => {
+				if (acc[`${d.clusterId}`]) {
+					acc[`${d.clusterId}`] = [...acc[`${d.clusterId}`], d]
+				} else {
+					acc[`${d.clusterId}`] = [d]
+				}
+				return acc
+			}, {})
+		}
+	}, [join])
+
+	const searchForChildren = useCallback(
+		(selection: string): JoinData[][] | undefined => {
+			if (clusterIdMap) {
+				return search([selection], clusterIdMap)
+			}
+		},
+		[clusterIdMap],
+	)
+
+	const selectedJoin = useMemo(
+		(): JoinData[][] | undefined => searchForChildren(selectedOption),
+		[searchForChildren, selectedOption],
+	)
+
+	const communityMap = useMemo((): { [x: string]: JoinData } | undefined => {
+		if (join) {
+			return join.reduce((acc: { [x: string]: JoinData }, d: JoinData) => {
+				if (acc[`${d.nodeId}`]) {
```

Comment:
what does this if statement actually do?

Reply:
This creates a map based on nodeId to look up on line 123 `const clusterId = ${communityMap[targetId].clusterId}` to avoid to a `.find` for each lookup for parent of edge. I iterate through selectedData and then find edges by id. Then use the id prop to look up in the nodeId map .

Reply:
This code assigns to acc[d.nodeId] on both branches, I don't think you need an `if` statement